ios/RNPdftron.m

```diff
@@ -109,6 +109,31 @@ - (void)setPassword:(NSString *)password onPDFDoc:(PTPDFDoc *)pdfDoc
     }
 }
 
+RCT_EXPORT_METHOD(pdfFromOfficeTemplate:(NSString *)docxPath json:(NSDictionary *)json resolver:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject)
+{
+    @try {
+        PTPDFDoc* pdfDoc = [[PTPDFDoc alloc] init];
+        NSData *jsonData = [NSJSONSerialization dataWithJSONObject:json options:0 error:nil];
+        NSString *jsonString = [[NSString alloc] initWithData:jsonData encoding:NSUTF8StringEncoding];
+        PTOfficeToPDFOptions* options = [[PTOfficeToPDFOptions alloc] init];
+        [options SetTemplateParamsJson:jsonString];
+        [PTConvert OfficeToPDF:pdfDoc in_filename:docxPath options:options];
+        
+        NSString* tempDirectory = NSTemporaryDirectory();
+        NSString* fileName = [NSUUID UUID].UUIDString;
+        NSString* resultPdfPath = [NSString stringWithFormat:@"%@%@%@", tempDirectory, fileName, @".pdf"];
```

Comment:
This will work, but check out the API

https://developer.apple.com/documentation/foundation/nsstring/1417069-stringbyappendingpathcomponent

NSString has a lot of good APIs for manipulation including when they represent paths. Generally using them is less error prone, and easier to read.